tests/test_transactions.py

```diff
@@ -166,3 +169,37 @@ def test_transaction_failure(
 
     # assert that no commit happens because of the exception.
     assert counter.count("commits_api.commit") == 0
+
+
+def test_placeholder_representations(
+    random_file_factory: RandomFileFactory,
+    fs: LakeFSFileSystem,
+    repository: str,
+    temp_branch: str,
+) -> None:
+    random_file = random_file_factory.make()
+
+    lpath = str(random_file)
+    rpath = f"{repository}/{temp_branch}/{random_file.name}"
+
+    message = f"Add file {random_file.name}"
+
+    with fs.transaction as tx:
+        fs.put_file(lpath, rpath)
+        sha = tx.commit(repository, temp_branch, message=message)
```

Comment:
Woah, that just hit me - `sha` will be `None` here in case of no unstaged changes, right? That's a bug.

Reply:
What is the desired behaviour? Do we want it to be a `Placeholder` instance or `Commit` (or whatever we are placeholding)?

Reply:
Actually, `client_helpers.commit` will return the tip of the branch if no unstaged changes exist, so this is not a bug.